### PR TITLE
Refactoring of the Effect type to RxSwift

### PR DESCRIPTION
**Pull request by ixrevo**
Hi Brandon and Stephen! As you [requested ](https://twitter.com/pointfreeco/status/1196463506058547200) the RxSwift support.

The first commit is a copy/paste of the original code. If you want me to restructure the code/files in the other way, just let me know.

I'm not sure if it's an idiomatic approach for RxSwift — conform a type to Observable protocol. Maybe, the better way is to use the Signal trait as `typealias` to the `Effect` which guarantees the following properties:

 - it never fails
 - it delivers events on `MainScheduler.instance`
 - `share(scope: .whileConnected)` sharing strategy

On the one hand, it quite matches the purpose of the `Effect` — to deliver `Action`s into the store. On the other hand, it makes it more opinionated.

### CODE REVIEW

**Comment by mbrandonw** [comment]
Awesome, I think this looks pretty good! I'm still digesting it, but one small thing: could you delete the `Vendor/swift-snapshot-testing` directory? That appears to have been mistakenly committed to our repo.

File: 0081-combine-and-effects-pt2/RxPrimeTime/ComposableArchitecture/ComposableArchitecture.swift [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  ) -> Store<LocalValue, LocalAction> {
+    let localStore = Store<LocalValue, LocalAction>(
+      initialValue: toLocalValue(self.value),
+      reducer: { localValue, localAction in
+        self.send(toGlobalAction(localAction))
+        localValue = toLocalValue(self.value)
+        return []
+    }
+    )
+    localStore.viewCancellable = self.$value.sink { [weak localStore] newValue in
```
[the remark this answers is not in the document]
Comment (mbrandonw) [reply 0081-combine-and-effects-pt2/RxPrimeTime/ComposableArchitecture/ComposableArchitecture.swift]: @haaakon I commented in your PR, but it is definitely ok to leave Combine here. We mostly want to concentrate on the effects side of things, not the SwiftUI side.

**Comment by mbrandonw** [comment]
Hey @ixrevo, I think this is good to merge, but I think the observable version is the one to go with and not the signal one. We rather not enforce effects to emit values on the main thread right now. That could change in the future, but just to keep things true to the current state of the architecture we think it'd be best to use a plain observable for now.

Thanks again!

**Comment by ixrevo** [comment]
> Hey @ixrevo, I think this is good to merge, but I think the observable version is the one to go with and not the signal one. We rather not enforce effects to emit values on the main thread right now. That could change in the future, but just to keep things true to the current state of the architecture we think it'd be best to use a plain observable for now.
> 
> Thanks again!

Hi @mbrandonw! No problem! I just thought it would be better to see the difference between these two approaches in the real code, not only in theory. Also, it helped me to spot the bug where I was creating the effect eagerly :)

**Pull request merged**